pallets/identity/src/lib.rs

```diff
@@ -1398,6 +1401,17 @@ impl<T: Trait> Module<T> {
         leeway: T::Moment,
         filter_cdd_id: Option<CddId>,
     ) -> Option<IdentityId> {
+        Self::base_fetch_valid_cdd_claims(claim_for, leeway, filter_cdd_id)
+            .into_iter()
+            .map(|id_claim| id_claim.claim_issuer)
+            .next()
+    }
+
+    pub fn base_fetch_valid_cdd_claims(
+        claim_for: IdentityId,
+        leeway: T::Moment,
+        filter_cdd_id: Option<CddId>,
+    ) -> Vec<IdentityClaim> {
```

Comment:
```suggestion
    ) -> impl Iterator<Item = IdentityClaim> {
```
We can be more lazy here and avoid collecting the entire vector.